File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/AdjustableDateOrRelativeDateSequence.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class AdjustableDateOrRelativeDateSequence : ISerialized
    {
        public AdjustableDateOrRelativeDateSequence(XmlNode xmlNode)
        {
            XmlNode adjustableDateNode = xmlNode.SelectSingleNode("adjustableDate");
            
            if (adjustableDateNode != null)
            {
                if (adjustableDateNode.Attributes["href"] != null || adjustableDateNode.Attributes["id"] != null) 
                {
                    if (adjustableDateNode.Attributes["id"] != null) 
                    {
                        adjustableDateIDRef_ = adjustableDateNode.Attributes["id"].Value;
                        AdjustableDate ob = new AdjustableDate(adjustableDateNode);
                        IDManager.SetID(adjustableDateIDRef_, ob);
                    }
                    else if (adjustableDateNode.Attributes["href"] != null)
                    {
                        adjustableDateIDRef_ = adjustableDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        adjustableDate_ = new AdjustableDate(adjustableDateNode);
                    }
                }
                else
                {
                    adjustableDate_ = new AdjustableDate(adjustableDateNode);
                }
            }
            
        
            XmlNode relativeDateSequenceNode = xmlNode.SelectSingleNode("relativeDateSequence");
            
            if (relativeDateSequenceNode != null)
            {
                if (relativeDateSequenceNode.Attributes["href"] != null || relativeDateSequenceNode.Attributes["id"] != null) 
                {
                    if (relativeDateSequenceNode.Attributes["id"] != null) 
                    {
                        relativeDateSequenceIDRef_ = relativeDateSequenceNode.Attributes["id"].Value;
                        RelativeDateSequence ob = new RelativeDateSequence(relativeDateSequenceNode);
                        IDManager.SetID(relativeDateSequenceIDRef_, ob);
                    }
                    else if (relativeDateSequenceNode.Attributes["href"] != null)
                    {
                        relativeDateSequenceIDRef_ = relativeDateSequenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        relativeDateSequence_ = new RelativeDateSequence(relativeDateSequenceNode);
                    }
                }
                else
                {
                    relativeDateSequence_ = new RelativeDateSequence(relativeDateSequenceNode);
                }
            }
            
        
        }
        
    
        #region AdjustableDate_
        private AdjustableDate adjustableDate_;
        public AdjustableDate AdjustableDate_
        {
            get
            {
                if (this.adjustableDate_ != null)
                {
                    return this.adjustableDate_; 
                }
                else if (this.adjustableDateIDRef_ != null)
                {
                    adjustableDate_ = IDManager.getID(adjustableDateIDRef_) as AdjustableDate;
                    return this.adjustableDate_; 
                }
                else
                {
                      return this.adjustableDate_; 
                }
            }
            set
            {
                if (this.adjustableDate_ != value)
                {
                    this.adjustableDate_ = value;
                }
            }
        }
        #endregion
        
        public string adjustableDateIDRef_ { get; set; }
        #region RelativeDateSequence_
        private RelativeDateSequence relativeDateSequence_;
        public RelativeDateSequence RelativeDateSequence_
        {
            get
            {
                if (this.relativeDateSequence_ != null)
                {
                    return this.relativeDateSequence_; 
                }
                else if (this.relativeDateSequenceIDRef_ != null)
                {
                    relativeDateSequence_ = IDManager.getID(relativeDateSequenceIDRef_) as RelativeDateSequence;
                    return this.relativeDateSequence_; 
                }
                else
                {
                      return this.relativeDateSequence_; 
                }
            }
            set
            {
                if (this.relativeDateSequence_ != value)
                {
                    this.relativeDateSequence_ = value;
                }
            }
        }
        #endregion
        
        public string relativeDateSequenceIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/AdjustableDateOrRelativeDateSequence.cpp
// AdjustableDateOrRelativeDateSequence.cpp 
#include "AdjustableDateOrRelativeDateSequence.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

AdjustableDateOrRelativeDateSequence::AdjustableDateOrRelativeDateSequence(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //adjustableDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* adjustableDateNode = xmlNode->FirstChildElement("adjustableDate");

   if(adjustableDateNode){adjustableDateIsNull_ = false;}
   else{adjustableDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- adjustableDateNode , address : " << adjustableDateNode << std::endl;
   #endif
   if(adjustableDateNode)
   {
      if(adjustableDateNode->Attribute("href") || adjustableDateNode->Attribute("id"))
      {
          if(adjustableDateNode->Attribute("id"))
          {
             adjustableDateIDRef_ = adjustableDateNode->Attribute("id");
             adjustableDate_ = boost::shared_ptr<AdjustableDate>(new AdjustableDate(adjustableDateNode));
             adjustableDate_->setID(adjustableDateIDRef_);
             IDManager::instance().setID(adjustableDateIDRef_,adjustableDate_);
          }
          else if(adjustableDateNode->Attribute("href")) { adjustableDateIDRef_ = adjustableDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { adjustableDate_ = boost::shared_ptr<AdjustableDate>(new AdjustableDate(adjustableDateNode));}
   }

   //relativeDateSequenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* relativeDateSequenceNode = xmlNode->FirstChildElement("relativeDateSequence");

   if(relativeDateSequenceNode){relativeDateSequenceIsNull_ = false;}
   else{relativeDateSequenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- relativeDateSequenceNode , address : " << relativeDateSequenceNode << std::endl;
   #endif
   if(relativeDateSequenceNode)
   {
      if(relativeDateSequenceNode->Attribute("href") || relativeDateSequenceNode->Attribute("id"))
      {
          if(relativeDateSequenceNode->Attribute("id"))
          {
             relativeDateSequenceIDRef_ = relativeDateSequenceNode->Attribute("id");
             relativeDateSequence_ = boost::shared_ptr<RelativeDateSequence>(new RelativeDateSequence(relativeDateSequenceNode));
             relativeDateSequence_->setID(relativeDateSequenceIDRef_);
             IDManager::instance().setID(relativeDateSequenceIDRef_,relativeDateSequence_);
          }
          else if(relativeDateSequenceNode->Attribute("href")) { relativeDateSequenceIDRef_ = relativeDateSequenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { relativeDateSequence_ = boost::shared_ptr<RelativeDateSequence>(new RelativeDateSequence(relativeDateSequenceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<AdjustableDate> AdjustableDateOrRelativeDateSequence::getAdjustableDate()
{
   if(!this->adjustableDateIsNull_){
        if(adjustableDateIDRef_ != ""){
             return boost::shared_static_cast<AdjustableDate>(IDManager::instance().getID(adjustableDateIDRef_));
        }else{
             return this->adjustableDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableDate>();
   }
}
boost::shared_ptr<RelativeDateSequence> AdjustableDateOrRelativeDateSequence::getRelativeDateSequence()
{
   if(!this->relativeDateSequenceIsNull_){
        if(relativeDateSequenceIDRef_ != ""){
             return boost::shared_static_cast<RelativeDateSequence>(IDManager::instance().getID(relativeDateSequenceIDRef_));
        }else{
             return this->relativeDateSequence_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RelativeDateSequence>();
   }
}
}

